sys/include/uri_parser.h

```diff
@@ -133,6 +143,36 @@ int uri_parser_process(uri_parser_result_t *result, const char *uri,
  */
 int uri_parser_process_string(uri_parser_result_t *result, const char *uri);
 
+/**
+ * @brief   Provides a list of URI query parameters from a given URI parser
+ *          result.
+ *
+ * @note    The function **DOES NOT** check for duplicate query parameters.
+ *
+ * @pre `uri_parsed != NULL`
+ * @pre `params != NULL` and all its elements are set to zero.
+ *
+ * @param[in] uri_parsed   A parsed URI result. Must not be NULL.
+ * @param[out] params      An array of @ref uri_parser_query_param_t.
+ *                         Must not be NULL and all zero-valued on call. Will be
+ *                         filled with the name-value-pairs in
+ *                         uri_parser_result_t::query of @p uri_parsed. If the
+ *                         number of query parameters in @p uri_parsed exceeds
+ *                         @p params_len, the list will be truncated and the
+ *                         function returns -2.
+ * @param[in] params_len   The length of @p params
+ *
+ * @return  number of filled entries in @p params on success. Might be 0 if
+ *          uri_parser_result_t::query is NULL.
+ * @return  -1 on parsing error.
+ * @return  -2 when the number of query parameters exceeds @p params_len.
+ *          In that case, the array is filled with the first @p params_len
+ *          name-value-pairs in uri_parser_result_t::query of @p uri_parsed.
+ */
+int uri_parser_split_query(const uri_parser_result_t *uri_parsed,
```

Comment:
Why use a `uri_parser_result_t *` here when you are only interested in the `query` element?
This function could be more generic: 
```suggestion
int uri_parser_split_query(const char *query, size_t query_len,
```

Reply:
Since then I need to also add verification of the query string to this function, which is already done in `uri_parser_process_string()`